pulsar-broker-common/src/main/java/org/apache/pulsar/broker/ServiceConfiguration.java

```diff
@@ -889,6 +889,16 @@ The delayed message index time step(in seconds) in per bucket snapshot segment,
             + " unacked messages than this percentage limit and subscription will not receive any new messages "
             + " until that subscription acks back `limit/2` messages")
     private double maxUnackedMessagesPerSubscriptionOnBrokerBlocked = 0.16;
+    @FieldContext(
+            category = CATEGORY_POLICIES,
+            doc = "If enabled subscriptions stores keys of messages which allows consumer not "
+                    + "to stuck in case it goes to recently assigned. The setting allows to overcome "
+                    + "situation when new KeyShared consumers will not get any messages until a consumer "
+                    + "that did get messages disconnects or acks/nacks some messages "
+                    + "The trade of is the need to track all the not acked messages in subscription"
+                    + "which could potentially lead to performance degradation and higher memory consumption"
+    )
+    private boolean rememberNotAckedMessagesKey = false;
```

Comment:
Thanks for the link! Will do

Reply:
Initiated the PIP https://github.com/apache/pulsar/pull/21615 and sent [Discussion] email to [dev@pulsar.apache.org](mailto:dev@pulsar.apache.org)
Please notify me in case I need to do some other actions which I did not noticed
Thank you!